doc/api/async_hooks.md

```diff
@@ -0,0 +1,532 @@
+# Async Hooks
+
+> Stability: 1 - Experimental
+
+The `async_hooks` module provides an API to register callbacks tracking the
+lifetime of asynchronous resources created inside a Node.js application.
+It can be accessed using:
+
+```js
+const async_hooks = require('async_hooks');
+```
+
+## Terminology
+
+An asynchronous resource represents an object with an associated callback.
+This callback may be called multiple times, for example, the `connection` event
+in `net.createServer`, or just a single time like in `fs.open`. A resource
+can also be closed before the callback is called. AsyncHooks does not
+explicitly distinguish between these different cases but will represent them
+as the abstract concept that is a resource.
+
+## Public API
+
+### Overview
+
+Following is a simple overview of the public API.
+
+```js
+const async_hooks = require('async_hooks');
+
+// Return the ID of the current execution context.
+const cid = async_hooks.currentId();
+
+// Return the ID of the handle responsible for triggering the callback of the
+// current execution scope to call.
+const tid = async_hooks.triggerId();
+
+// Create a new AsyncHook instance. All of these callbacks are optional.
+const asyncHook = async_hooks.createHook({ init, before, after, destroy });
+
+// Allow callbacks of this AsyncHook instance to call. This is not an implicit
+// action after running the constructor, and must be explicitly run to begin
+// executing callbacks.
+asyncHook.enable();
+
+// Disable listening for new asynchronous events.
+asyncHook.disable();
+
+//
+// The following are the callbacks that can be passed to createHook().
+//
+
+// init() is called during object construction. The resource may not have
+// completed construction when this callback runs, therefore all fields of the
+// resource referenced by "id" may not have been populated.
+function init(id, type, triggerId, resource) { }
+
+// before() is called just before the resource's callback is called. It can be
+// called 0-N times for handles (e.g. TCPWrap), and will be called exactly 1
+// time for requests (e.g. FSReqWrap).
+function before(id) { }
+
+// after() is called just after the resource's callback has finished.
+function after(id) { }
+
+// destroy() is called when an AsyncWrap instance is destroyed.
+function destroy(id) { }
+```
+
+#### `async_hooks.createHook(callbacks)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `callbacks` {Object} the callbacks to register
+* Returns: `{AsyncHook}` instance used for disabling and enabling hooks
```

Comment:
I will add it in the `enable()` section. Unfortunately, there is limited space on this line.